pinball/ui/views.py

```diff
@@ -359,7 +359,7 @@ def signin(request):
 def auth_return(request):
     oauth2_flow = oauth2.OAuth2Flow()
     domains = oauth2_flow.get_domains()
-    flow = oauth2.OAuth2Flow().get_flow(domains[1])
+    flow = oauth2.OAuth2Flow().get_flow(oauth2_flow.domain)
```

Comment:
The second step (exchange token) does not need domain specific flow object. So changing it to domain[0] will solve your problem.